### DISCUSSION TITLE
Can a Basemodel have a field named schema?

### DISCUSSION THREAD

**Discussion by tekumara** [pydantic/pydantic#4718] [Question]
This would help model a database table, eg:
```python
from pydantic import BaseModel

class Table(BaseModel):
    database: str
    schema: str
    name: str
```    

The challenge is I'm getting the vscode pylance error:
```
"schema" overrides method of same name in class "BaseModel" with incompatible type "str"
```

**Comment by samuelcolvin** [comment]
ye, you can't use the name `schema` because of the `schema()` method on models by default, you'll need to use an alias, see https://pydantic-docs.helpmanual.io/usage/schema/#field-customization.

You'll want

```py
from pydantic import BaseModel, Field

class Table(BaseModel):
    database: str
    schema_: str = Field(alias='schema')
    name: str
```

**Reply by tekumara** [reply]
Thank you that could work, although it does make my model's api a little confusing (I can imagine users incorrectly using `schema` instead of `schema_`). Maybe I can use a dataclass instead.

**Reply by samuelcolvin** [reply]
This will change in V2, where all these methods will have a `model_` prefix.

**Reply by rrauenza** [reply]
Did schema get fixed?  In pydantic 2.4.2 I'm still seeing an issue with using 'schema' as a field:

`config.py:29: error: Incompatible types in assignment (expression has type "str", base class "BaseModel" defined the type as "Callable[[type[BaseModel], bool, str], dict[str, Any]]")  [assignment]
`

**Reply by quic-sponsfre** [reply]
Same here, any update on this?
